python/pip_install/pip_requirements.bzl

```diff
@@ -0,0 +1,68 @@
+"Rules to verify and update pip-compile locked requirements.txt"
+
+load("//python:defs.bzl", "py_binary", "py_test")
+load("//python/pip_install:repositories.bzl", "requirement")
+
+def pip_requirements(
+        name = "requirements_test",
+        extra_args = [],
+        **kwargs):
+    """
+    Produce two targets for checking pip-compile:
+
+    - validate with `bazel test some_test`
+    - update with   `bazel run some.update`
+
+    Args:
+        name: string
+        extra_args: passed to pip-compile
+        **kwargs: other bazel attributes
+    """
+    if name and not name.endswith("_test"):
+        fail("test target must be named with _test suffix")
+    requirements_in = name[:-5] + ".in"
+    requirements_txt = name[:-5] + ".txt"
+
+    data = kwargs.pop("data", []) + [requirements_in, requirements_txt]
+
+    loc = "$(rootpath %s)"
+
+    # Use the Label constructor so this is expanded in the context of the file
+    # where it appears, which is to say, in @rules_python
+    pip_compile = Label("//python/pip_install:pip_compile.py")
+
+    py_binary(
+        name = name[:-5] + ".update",
+        srcs = [pip_compile],
+        main = pip_compile,
+        args = [
+            loc % requirements_in,
+            loc % requirements_txt,
+        ] + extra_args,
+        visibility = ["//visibility:public"],
```

Comment:
Do both gen'd targets need public visibility? Which other targets in a WORKSPACE would depend on these targets? It seems like both could be default private and the macro could [conventionally accept defaulted `visibility` arg](https://docs.bazel.build/versions/master/skylark/macros.html#conventions)

Reply:
yeah I think this was copy-pasta, replaced with a visibilty arg defaulting to private. I don't see any case where you'd want to depend on either generated target